contracts/epochs/FlowEpoch.cdc

```diff
@@ -232,22 +232,55 @@ pub contract FlowEpoch {
     /// Metadata that is managed by the smart contract
     /// and cannot be changed by the Admin
 
+    /// Contains a historical record of the metadata from all previous epochs
+    /// indexed by epoch number
+
+    /// Returns the metadata from the specified epoch
+    /// or nil if it isn't found
+    /// Epoch Metadata is stored in account storage so the growing dictionary
+    /// does not have to be loaded every time the contract is loaded
+    pub fun getEpochMetadata(_ epochCounter: UInt64): EpochMetadata? {
+        if let metadataDictionary = self.account.copy<{UInt64: EpochMetadata}>(from: self.metadataStoragePath) {
+            return metadataDictionary[epochCounter]
+        }
+        return nil
+    }
+
+    /// Saves a modified EpochMetadata struct to the metadata in account storage
+    /// 
+    access(contract) fun saveEpochMetadata(_ newMetadata: EpochMetadata) {
+        pre {
+            self.currentEpochCounter == (0 as UInt64) ||
```

Comment:
Just FYI: With https://github.com/onflow/cadence/pull/957 (to be released, i.e. coming in Cadence v0.17) all those static casts shouldn't be necessary anymore, so this can become more readable